clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2516,6 +2604,29 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
     return true;
   }
 
+  // Default inits the type, then calls the init-method in the body
+  void handleSyclKernelHandlerType(ParmVarDecl *KernelHandlerArg) {
+
+    // Create and default initialize local clone of kernel handler
+    createKernelHandlerClone(SemaRef.getASTContext(),
+                             DeclCreator.getKernelDecl(), KernelHandlerArg);
+
+    // Add declaration statement to openCL kernel body
+    Stmt *DS =
+        new (SemaRef.Context) DeclStmt(DeclGroupRef(KernelHandlerClone),
+                                       KernelCallerSrcLoc, KernelCallerSrcLoc);
+    BodyStmts.push_back(DS);
+
+    // Generate
+    // KernelHandlerClone.__init_specialization_constants_buffer(specialization_constants_buffer)
+    // call if target does not have native support for specialization constants.
+    // Here, specialization_constants_buffer is the compiler generated kernel
+    // argument of type char*.
+    llvm::Triple T = SemaRef.Context.getTargetInfo().getTriple();
+    if (!(T.isSPIR() && T.getSubArch() == llvm::Triple::NoSubArch))
```

Comment:
Seen this 2x now... I wonder if there is a good single-name for this check we could give it as its own function (if only to better document what it means).

Reply:
I could move it to a function hasNativeSycl2020SpecConstantSupport(). I agree this improves readability. Thanks for the suggestion!